Mathlib/Computability/NFA.lean

```diff
@@ -102,17 +109,36 @@ theorem evalFrom_cons (S : Set σ) (a : α) (x : List α) :
     M.evalFrom S (a :: x) = M.evalFrom (M.stepSet S a) x :=
   rfl
 
+variable (M) in
 @[simp]
+theorem evalFrom_append (S : Set σ) (x y : List α) :
+    M.evalFrom S (x ++ y) = M.evalFrom (M.evalFrom S x) y := by
+  simp only [evalFrom, List.foldl_append]
+
 theorem evalFrom_append_singleton (S : Set σ) (x : List α) (a : α) :
     M.evalFrom S (x ++ [a]) = M.stepSet (M.evalFrom S x) a := by
```

Comment:
Maybe this lemma can be deprecated in favor of `evalFrom_append`?

Reply:
I'm ok with that, should I also remove `eval_append_singleton` (when I remove these everything still builds it seems)?

In these cases, does Mathlib generally prefer to remove such lemmas, or would it be still useful to keep them around as a shorthand?

Thanks!

Reply:
You should deprecate them indeed